lib/node.js

```diff
@@ -501,7 +544,38 @@ declare module "tls" {
 }
 
 declare module "url" {
-  // TODO
+  declare function parse(
+    urlStr: string,
+    parseQueryString?: boolean,
+    slashesDenoteHost?: boolean
+  ): {
+    protocol: ?string;
+    slashes: ?string;
+    auth: ?string;
+    host: ?string;
+    port: ?string;
+    hostname: ?string;
+    hash: ?string;
+    search: ?string;
+    query: ?any; // null | string | Object
+    pathname: string;
+    path: string;
+    href: string;
+  };
+  declare function format(urlObj: {
+    href?: string;
```

Comment:
They look similar but they're subtly different. (1) `format` doesn't take a `path` prop, (2) and`format`'s  `query` can only be an ?Object. (3) Unlike `parse`'s `query` that can be `null | string | Object`.

I've fixed the rest of the nullable vs optional.